python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -81,8 +81,13 @@ def handle_messages(self) -> Iterator[PipesParams]:
         with self._message_reader.read_messages(self) as params:
             yield params
 
-    def get_reported_results(self) -> Sequence[PipesExecutionResult]:
-        return tuple(self._result_queue.queue)
+    def clear_result_queue(self) -> Iterator[PipesExecutionResult]:
+        while not self._result_queue.empty():
+            yield self._result_queue.get()
+
+    def __del__(self):
+        if not self._result_queue.empty():
+            self._context.log.warn("[pipes] some reported results were not processed.")
```

Comment:
since we lose the session context manager to "complete" the external invocation we need some other hook for when to know the op/asset was closed without processing this.  

Initial impl with the ol cursed `__del__` here. One option I considered was adding the capability to register callbacks on the `context` to run when the op/asset function is over.